bin/python/ripple/util/Sign.py

```diff
@@ -0,0 +1,139 @@
+#!/usr/bin/env python
+
+from __future__ import print_function
+
+import base64, os, random, sys
+import ed25519
+import ecdsa
+from ripple.util import Base58
+from ripple.ledger import SField
+
+ED_BYTE = chr(0xed)
+
+USAGE = """\
+Usage:
+    create
+        Create a new master public/secret key pair.
+
+    check <key>
+        Check an existing key for validity.
+
+    sign <sequence> <validator-public> <master-secret>
+        Create a new signed manifest with the given sequence
+        number, validator public key, and master secret key.
+    """
+
+def to_bytes_bigend(n, length):
+    h = '%x' % n
+    return ('0' * (len(h) % 2) + h).zfill(length * 2).decode('hex')
+
+def prepend_length_byte(b):
+    assert len(b) < 192, 'Too long'
+    return chr(len(b)) + b
+
+#-----------------------------------------------------------
+
+def make_seed(urandom=os.urandom):
+    # This is not used.
+    return urandom(16)
+
+def make_ed25519_keypair(urandom=os.urandom):
+    private_key = urandom(32)
+    return private_key, ed25519.publickey(private_key)
+
+def make_ecdsa_keypair():
```

Comment:
According to Python docs `urandom` will “[r]eturn a string of `n` random bytes suitable for cryptographic use.” While it does use `/dev/urandom` instead of `/dev/random` it _should_ be fine, even for long-term cryptographic key generation unless the available entropy in the pool is very low.